#161255203 Users should be able to tag an article

Tony-Ndichu opened the pull request:
#### What does this PR do?

This PR enables users to tag an article

#### Description of Task to be completed?

Users should be able to add tags while creating articles and view those tags when viewing an article. 

#### How should this be manually tested?

- After cloning the repo, cd into it
- Checkout the branch `ft-tag-an-article`
- Run `npm install`
- Run `npm start`

#### What are the relevant pivotal tracker stories?
[#161255203](https://www.pivotaltracker.com/story/show/161255203)

#### Screenshots (if appropriate):
When posting an article:

<img width="1440" alt="screenshot 2018-12-19 at 23 29 42" src="https://user-images.githubusercontent.com/37045046/50246599-e3c1c880-03e6-11e9-9a18-6f207426fcea.png">
<img width="1440" alt="screenshot 2018-12-19 at 23 29 56" src="https://user-images.githubusercontent.com/37045046/50246620-f1774e00-03e6-11e9-9fa3-0eb3b2605944.png">
<img width="1440" alt="screenshot 2018-12-19 at 23 30 12" src="https://user-images.githubusercontent.com/37045046/50246641-fd631000-03e6-11e9-9314-1c5fc32e8142.png">

When viewing an article:
<img width="1440" alt="screenshot 2018-12-19 at 23 33 57" src="https://user-images.githubusercontent.com/37045046/50246732-30a59f00-03e7-11e9-9b1d-943b5b61fd7d.png">
<img width="1440" alt="screenshot 2018-12-19 at 23 30 50" src="https://user-images.githubusercontent.com/37045046/50246712-1e2b6580-03e7-11e9-851b-1b340dea8a42.png">

Tony-Ndichu committed:
feat(tags): users should be able to tag an article

- add tag selection while creating an article
- display tags on viewing article

[WIP #161255203]

Tony-Ndichu committed:
feat(tags): users should be able to tag an article 

- add tag selection while creating an article
- display tags on viewing article

[Finishes #161255203]

Rodgers-M commented on .codeclimate.yml:
```diff
@@ -17,7 +17,7 @@ checks:
   method-lines:
     config:
-      threshold: 70
+      threshold: 100
```
why do you need so many lines per method? if a method grows so big then it is an indication its doing a lot and needs to be broken down

Tony-Ndichu replied:
On it

Tony-Ndichu replied:
@Rodgers-M , with our current application design and setup, this is the component where all the different functions that affect creating an article are defined and later on passed as props, its a flow we had decided on rather than having the different submit functions in separate places, it would make it hard for someone new to this codebase. Also as it is, it allows us to easily plugin different features to the articles CRUD e.g this tag feature that is used when an article is being created.

Rodgers-M commented on src/containers/Article/index.js:
```diff
@@ -61,6 +65,14 @@ export class ArticleForm extends Component {
     this.setState({ body: blocks });
   };
 
+  handleTags = (newValue) => {
```
what should `newValue` contain? please make it more descriptive

Tony-Ndichu replied:
On it

Rodgers-M commented on src/containers/Article/index.js:
```diff
@@ -61,6 +65,14 @@ export class ArticleForm extends Component {
 
+  handleTags = (newValue) => {
+    const tags = [];
+    newValue.map((value) => {
```
same case here, the name `newValue` is singular, but again you are mapping over it and expecting a value!!!! if its an array, the name should be pluralized

Tony-Ndichu replied:
On it too

Rodgers-M commented on src/containers/Article/index.js:
```diff
@@ -61,6 +65,14 @@ export class ArticleForm extends Component {
 
+  handleTags = (tags) => {
+    const tagArray = [];
+    tags.map((value) => {
```
it's a good practice to name your function arguments appropriately. In this case, if I have an array of tags and mapping over it, am expecting to be getting a tag at each iteration, having the name as value is ambiguous. Just for best practices its better doing this

`tags.map((tag) =>{....}`

like in line 71, `value.value` is misleading , 

> object `value` has a property `value`!!!

Tony-Ndichu replied:
Got it , let me do it

Rodgers-M commented on src/containers/Article/index.js:
```diff
@@ -71,14 +83,17 @@ export class ArticleForm extends Component {
       formData.append('title', title);
       formData.append('body', JSON.stringify(body));
       formData.append('image_file', imageFile);
+      tags.map((value) => {
```
mapping over tags here too

Tony-Ndichu replied:
Got it

Tony-Ndichu committed:
feat(tags): users should be able to tag an article

- add tag selection while creating an article
- display tags on viewing article

[Finishes #161255203]
